### PR TITLE
Solve #109 problem

### PR DESCRIPTION
**Pull request by EstelleDa**

### CODE REVIEW

File: src/components/screens/ScoreSetEditor.vue [hunk 1]
```diff
@@ -1282,7 +1284,27 @@ export default {
         this.secondaryPublicationIdentifiers = []
         this.publicationIdentifiers = []
         this.dataUsagePolicy = null
-        this.targetGene = null
+        this.existingTargetGene = null
+        this.targetGene = {
```
Comment (ashsny) [inline src/components/screens/ScoreSetEditor.vue:1288]: this is redundant with l686 where target gene is created. I think you want to add a helper function emptyTargetGene outside the component to wrap this code and then find all the similar code and replace it with this.

File: src/components/screens/ScoreSetEditor.vue [hunk 2]
```diff
@@ -790,13 +790,15 @@ export default {
     'targetGene.externalIdentifiers': {
       deep: true,
       handler: function (newValue) {
-        // If an identifier has been set, set the offset to 0 by default.
-        for (const dbName of externalGeneDatabases) {
-          if (newValue[dbName]?.identifier?.identifier != null && newValue[dbName]?.offset == null) {
-            this.targetGene.externalIdentifiers[dbName].offset = 0
+        if (newValue) {
```
Comment (ashsny) [inline src/components/screens/ScoreSetEditor.vue:793]: instead of wrapping the whole function in `if (newValue) {`
just add 

```
if (!newValue) {
  return
}
```

File: src/components/screens/ScoreSetEditor.vue [hunk 3]
```diff
@@ -1291,7 +1313,6 @@ export default {
         this.geneName = null
         this.geneNameAccessionSuggestions = []
         this.geneNameDropdownValue =  null
-        this.targetGenes = []
```
Comment (ashsny) [inline src/components/screens/ScoreSetEditor.vue:1294]: why is this deleted? what is the effect of not clearing this on clear()?

File: src/components/screens/ScoreSetEditor.vue (hunk 1 above)
Comment (EstelleDa) [reply src/components/screens/ScoreSetEditor.vue:1288]: Good idea.

File: src/components/screens/ScoreSetEditor.vue (hunk 3 above)
Comment (EstelleDa) [reply src/components/screens/ScoreSetEditor.vue:1294]: It's repeated. Another `this.targetGenes = []` is in line 1308.

File: src/components/screens/ScoreSetEditor.vue [hunk 4]
```diff
@@ -1282,22 +1285,21 @@ export default {
...
+      this.assembly = null
+      this.assemblySuggestions = []
+      this.assemblyDropdownValue = null
+      this.existingTargetGene = null
+      this.geneName = null
+      this.geneNameAccessionSuggestions = []
+      this.geneNameDropdownValue =  null
       this.fileCleared('targetGeneTargetSequenceSequenceFile')
+      this.referenceGenome = null
       this.targetGene = {
```
Comment (ashsny) [inline src/components/screens/ScoreSetEditor.vue:1303]: please also fix redundancy with l686 by adding helper function returning this target gene

Comment (ashsny) [reply src/components/screens/ScoreSetEditor.vue:1289]: it looks like you're not yet using the helper function for the initialization at l686.

File: src/components/screens/ScoreSetEditor.vue (hunk 3 above)
Comment (ashsny) [reply src/components/screens/ScoreSetEditor.vue:1294]: this.targetGenes is no longer getting cleared in the 'New score set' case when clear is clicked due to this deletion. not sure what you mean by it's repeated here, I don't see it on l1308 before or after your changes.

**Comment by ashsny** [comment]
I'll finish this up as I want it to go in with the frontend push today.

File: src/components/screens/ScoreSetEditor.vue (hunk 3 above)
Comment (EstelleDa) [reply src/components/screens/ScoreSetEditor.vue:1294]: Maybe the repeated codes were from my other branch.

File: src/components/screens/ScoreSetEditor.vue (hunk 4 above)
Comment (EstelleDa) [reply src/components/screens/ScoreSetEditor.vue]: Thanks Ashley! Now I know what is the helper function.
